docs/python/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+###############################################################################
+# Top contributors (to current version):
+#   Gereon Kremer
+#
+# This file is part of the cvc5 project.
+#
+# Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+# in the top-level source directory and their institutional affiliations.
+# All rights reserved.  See the file COPYING in the top-level source
+# directory for licensing information.
+# #############################################################################
+#
+# Build system configuration for python API documentation.
+##
+
+add_custom_target(docs-python)
+
+if (BUILD_BINDINGS_PYTHON)
+    # Python API docs are generated from built python API
+    add_dependencies(docs-python pycvc5)
```

Comment:
Just for my own understanding: is the documentation actually generated from the `pycvc5` python module itself? As in it exercises the python bindings to get the documentation? Or is it just that the `pycvc5` documentation is generated within the `pycvc5` target in CMake? If this question doesn't make any sense, no worries about answering -- I was just curious.

Reply:
Yes. The easiest way to generate documentation for python code with Sphinx is using `autodoc`, which will essentially do introspection on the python module and extract the docstrings. For this to work, the module needs to be working.